src/zenml/mlstacks/utils.py

```diff
@@ -0,0 +1,525 @@
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def stack_exists(stack_name: str) -> bool:
+    """Checks whether a stack with that name exists or not.
+
+    Args:
+        stack_name: The name of the stack to check for.
+
+    Returns:
+        A boolean indicating whether the stack exists or not.
+    """
+    try:
+        Client().get_stack(
+            name_id_or_prefix=stack_name, allow_name_prefix_match=False
+        )
+    except KeyError:
+        return False
+    return True
+
+
+@verify_installation
+def stack_spec_exists(stack_name: str) -> bool:
+    """Checks whether a stack spec with that name exists or not.
+
+    Args:
+        stack_name: The name of the stack spec to check for.
+
+    Returns:
+        A boolean indicating whether the stack spec exists or not.
+    """
+    from mlstacks.constants import MLSTACKS_PACKAGE_NAME
+
+    spec_dir = (
+        f"{click.get_app_dir(MLSTACKS_PACKAGE_NAME)}/stack_specs/{stack_name}"
+    )
+    return Path(spec_dir).exists()
+
+
+def get_mlstacks_version() -> Optional[str]:
+    """Gets the version of mlstacks locally installed.
+
+    Raises:
+        pkg_resources.DistributionNotFound: If mlstacks is not installed.
+    """
+    try:
+        return pkg_resources.get_distribution(
+            STACK_RECIPE_PACKAGE_NAME
+        ).version
+    except pkg_resources.DistributionNotFound:
+        return
+
+
+def _get_component_flavor(
+    key: str, value: Union[bool, str], provider: str
+) -> str:
+    """Constructs the component flavor from Click CLI params.
+
+    Args:
+        key: The component key.
+        value: The component value.
+        provider: The provider name.
+
+    Returns:
+        The component flavor.
+    """
+    if key in {"artifact_store", "container_registry"} and bool(value):
+        if provider == "aws":
+            flavor = "s3"
+        elif provider == "azure":
+            flavor = "azure"
+        elif provider == "gcp":
+            flavor = "gcp"
+    elif (
+        key
+        in {
+            "experiment_tracker",
+            "orchestrator",
+            "model_deployer",
+            "model_registry",
+            "mlops_platform",
+            "step_operator",
+        }
+        and value
+        and type(value) == str
+    ):
+        flavor = value
+    return flavor
+
+
+@verify_installation
+def _add_extra_config_to_components(
+    components: List["Component"], extra_config: Dict[str, str]
+) -> List["Component"]:
+    """Adds extra config to mlstacks `Component` objects.
+
+    Args:
+        components: A list of mlstacks `Component` objects.
+        extra_config: A dictionary of extra config.
+
+    Returns:
+        A list of mlstacks `Component` objects.
+    """
+    from mlstacks.models.component import ComponentMetadata
+
+    # Define configuration map
+    config_map = {
+        ("container_registry",): ["repo_name"],
+        ("artifact_store",): ["bucket_name"],
```

Comment:
Is it provider agnostic? Again thinking of `k3d`

Reply:
Not sure I understand. Is what provider-agnostic? artifact-store + providing the bucket name?